lib/components/user-input/MultiSelect.tsx

```diff
@@ -0,0 +1,155 @@
+import { useState } from 'react'
+import { Search } from 'lucide-react'
+import { useTranslation } from '../../hooks/useTranslation'
+import type { PropsWithLanguage } from '../../hooks/useTranslation'
+import type { Languages } from '../../hooks/useLanguage'
+import { MultiSearchWithMapping } from '../../util/simpleSearch'
+import { tx, tw } from '../../twind'
+import { Span } from '../Span'
+import { Menu, MenuItem } from './Menu'
+import { Input } from './Input'
+import { Checkbox } from './Checkbox'
+
+type MultiSelectTranslation = {
+  select: string,
+  search: string,
+  selected: string
+}
+
+const defaultMultiSelectTranslation: Record<Languages, MultiSelectTranslation> = {
+  en: {
+    select: 'Select',
+    search: 'Search',
+    selected: 'selected'
+  },
+  de: {
+    select: 'Auswählen',
+    search: 'Suche',
+    selected: 'ausgewählt'
+  }
+}
+
+// TODO maybe add custom item builder here
+export type MultiSelectOption<T> = {
+  label: string,
+  value: T,
+  selected: boolean,
+  disabled?: boolean,
+  className?: string
+}
+
+export type SearchProps<T> = {
+  initialSearch?: string,
+  searchMapping: (value: MultiSelectOption<T>) => string[]
+}
+
+export type MultiSelectProps<T> = {
+  options: MultiSelectOption<T>[],
+  onChange: (options: MultiSelectOption<T>[]) => void,
+  search?: SearchProps<T>,
+  disabled?: boolean,
+  selectedDisplay?: (props: {
+    items: MultiSelectOption<T>[],
+    disabled: boolean
+  }) => ReactNode,
+  label?: string,
+  hintText?: string,
+  showDisabledOptions: boolean,
+  className: string,
+  labelClassName: string
+}
+
+/**
+ * A Component for multi selction
+ */
+export const MultiSelect = <T, >({
+  language,
+  options,
+  onChange,
+  search,
+  disabled = false,
+  selectedDisplay,
+  label,
+  hintText,
+  showDisabledOptions = true,
+  className,
+  labelClassName = 'text-lg font-semibold',
+}: PropsWithLanguage<MultiSelectTranslation, MultiSelectProps<T>>) => {
+  const translation = useTranslation(language, defaultMultiSelectTranslation)
+  const [searchText, setSearchText] = useState<string>(search?.initialSearch ?? '')
+  let filteredOptions: MultiSelectOption<T>[] = options
+  const enableSearch = !!search
+  if (enableSearch && !!searchText) {
+    filteredOptions = MultiSearchWithMapping<MultiSelectOption<T>>(
+      searchText,
+      filteredOptions,
+      value => search.searchMapping(value))
+  }
+  if (!showDisabledOptions) {
+    filteredOptions = filteredOptions.filter(value => !value.disabled)
+  }
+
+  const selectedItems = options.filter(value => value.selected)
+  const menuButtonText = selectedItems.length === 0 ?
+    hintText ?? translation.select
+    : <Span>{`${selectedItems.length} ${translation.selected}`}</Span>
+
+  return (
+    <div className={tx(className)}>
+      {label && (
+        <label htmlFor={label} className={tx(' mb-1', labelClassName)}>
+          {label}
+        </label>
+      )}
+      <Menu<HTMLDivElement>
+        alignment="t_"
```

Comment:
Were is the closing generic?

Reply:
You cannot close with a generic, you close with </Menu> at least according to my linter